shared/teams/team/member/add-to-channels.tsx

```diff
@@ -0,0 +1,220 @@
+            numMembers={item.numMembers}
+            selected={selected.has(item.channelname)}
+            onSelect={() => onSelect(item.channelname)}
+          />
+        )
+    }
+  }
+
+  return (
+    <Kb.Modal
+      header={{
+        hideBorder: Styles.isMobile,
+        leftButton: Styles.isMobile ? <Kb.Text type="BodyBigLink">Cancel</Kb.Text> : undefined,
+        rightButton: Styles.isMobile ? (
+          <Kb.Text type="BodyBigLink" onClick={onFinish} style={!numSelected && styles.disabled}>
+            Add
+          </Kb.Text>
+        ) : (
+          undefined
+        ),
+        title: <ModalTitle teamname={meta.teamname} title={`Add ${username} to...`} />,
+      }}
+      footer={
+        Styles.isMobile
+          ? undefined
+          : {
+              content: (
+                <Kb.Box2 direction="horizontal" gap="tiny" fullWidth={true}>
+                  <Kb.Button
+                    type="Dim"
+                    label="Cancel"
+                    onClick={onCancel}
+                    style={Styles.globalStyles.flexOne}
+                  />
+                  <Kb.Button
+                    label={
+                      numSelected ? `Add to ${numSelected} ${pluralize('channel', numSelected)}` : 'Add...'
+                    }
+                    onClick={onFinish}
+                    disabled={!numSelected}
+                    style={Styles.globalStyles.flexOne}
+                  />
+                </Kb.Box2>
+              ),
+            }
+      }
+      allowOverflow={true}
+      noScrollView={true}
+    >
+      <Kb.Box2 direction="vertical" fullWidth={true} style={Styles.globalStyles.flexOne}>
+        <Kb.Box2 direction="horizontal" fullWidth={true} style={styles.searchFilterContainer}>
+          <Kb.SearchFilter
+            placeholderText={`Search ${channelInfos.size} ${pluralize('channel', channelInfos.size)}`}
+            icon="iconfont-search"
+            onChange={setFilter}
+            size="full-width"
+            onFocus={() => setFiltering(true)}
+            onBlur={() => setFiltering(false)}
+          />
+        </Kb.Box2>
+        <Kb.Box2 direction="vertical" style={styles.listContainer} fullWidth={true}>
+          <Kb.List2
+            items={items}
+            renderItem={renderItem}
+            itemHeight={
+              Styles.isMobile ? {height: 48, type: 'fixed'} : {sizeType: 'Large', type: 'fixedListItem2Auto'}
+            }
+          />
+        </Kb.Box2>
+      </Kb.Box2>
+    </Kb.Modal>
+  )
+}
+
+const HeaderRow = ({onCreate, onSelectAll, onSelectNone}) => (
+  <Kb.Box2
+    direction="horizontal"
+    alignItems="center"
+    fullWidth={true}
+    style={Styles.collapseStyles([styles.item, styles.headerItem])}
+  >
+    <Kb.Button label="Create channel" small={true} mode="Secondary" onClick={onCreate} />
+    <Kb.Text type="BodyPrimaryLink" onClick={onSelectAll || onSelectNone}>
+      {onSelectAll ? 'Select all' : 'Select none'}
```

Comment:
Is `select none` from design? "Clear" or "Clear selection" feels more natural/common to me.

Reply:
I changed it to "Clear"